### PR TITLE
Rework DDL operators

### PR DESCRIPTION
**Pull request by ray6080**
Merge DDL operators into three:
- Alter
- CreateTable
- DropTable

Operations around `comment` should also be refactored into `Alter` operator.

Note: We didn't correctly handle case-insensitive before this change. Now it should be correctly done. Need more tests on this.

### CODE REVIEW

File: src/storage/wal/wal.cpp [hunk 1]
```diff
@@ -160,10 +153,9 @@ void WAL::addNewWALRecordNoLock(WALRecord& walRecord) {
 void WAL::setIsLastRecordCommit() {
     WALIterator walIterator(fileHandle, mtx);
     WALRecord walRecord;
+    KU_ASSERT(walIterator.hasNextRecord());
     if (!walIterator.hasNextRecord()) {
-        logger->info(
-            "Opening an existing WAL file but the file is empty. This should never happen. file: " +
-            fileHandle->getFileInfo()->path);
+        // Opening an existing WAL file but the file is empty. This should never happen.
```
Comment (andyfengHKU) [inline src/storage/wal/wal.cpp:158]: I wonder if we should assert for a behaviour that will never happen.

File: src/include/binder/binder.h [hunk 2]
```diff
@@ -15,9 +16,13 @@ struct CreateTableInfo;
 struct BaseScanSource;
 } // namespace parser
 
+namespace catalog {
```
Comment (andyfengHKU) [inline src/include/binder/binder.h:19]: Do we need this?

File: src/include/binder/binder.h [hunk 3]
```diff
@@ -26,10 +31,10 @@ class Database;
 
 namespace function {
 struct TableFunction;
-}
+} // namespace function
 
 namespace binder {
-
+struct PropertyInfo;
```
Comment (andyfengHKU) [inline src/include/binder/binder.h:37]: ditto

File: src/include/catalog/catalog_content.h [hunk 4]
```diff
@@ -74,20 +77,25 @@ class CatalogContent {
...
-    common::table_id_t createNodeTable(const binder::BoundCreateTableInfo& info);
-    common::table_id_t createRelTable(const binder::BoundCreateTableInfo& info);
-    common::table_id_t createRelGroup(const binder::BoundCreateTableInfo& info);
-    common::table_id_t createRDFGraph(const binder::BoundCreateTableInfo& info);
+    common::table_id_t createTable(const binder::BoundCreateTableInfo& info);
     void dropTable(common::table_id_t tableID);
+    void alterTable(const binder::BoundAlterInfo& info);
+
+    // ----------------------------- Internal functions -----------------------
+    std::unique_ptr<CatalogEntry> createNodeTableEntry(
```
Comment (andyfengHKU) [inline src/include/catalog/catalog_content.h:85]: try make private if these suppose to be internal functions.

**Review by andyfengHKU — APPROVED** [review]

File: src/catalog/catalog_content.cpp [hunk 5]
```diff
@@ -48,13 +73,11 @@ table_id_t CatalogContent::createNodeTable(const binder::BoundCreateTableInfo& i
...
-    tableNameToIDMap.emplace(nodeTableEntry->getName(), tableID);
-    tables->createEntry(std::move(nodeTableEntry));
-    return tableID;
+    return nodeTableEntry;
 }
 
-table_id_t CatalogContent::createRelTable(const binder::BoundCreateTableInfo& info) {
-    table_id_t tableID = assignNextTableID();
+std::unique_ptr<CatalogEntry> CatalogContent::createRelTableEntry(
+    common::table_id_t tableID, const binder::BoundCreateTableInfo& info) {
```
Comment (acquamarin) [inline src/catalog/catalog_content.cpp:80]: using namespace kuzu::binder;
using namespace kuzu::common;
using namespace kuzu::storage;
I see these on the top of the file, so you can remove binder/common namespace there.

File: src/catalog/catalog_content.cpp [hunk 6]
```diff
@@ -135,14 +150,57 @@ void CatalogContent::dropTable(common::table_id_t tableID) {
...
+    }
+    }
+}
+
 void CatalogContent::renameTable(table_id_t tableID, const std::string& newName) {
+    // TODO(Xiyang/Ziyi): Do we allow renaming of rel table groups?
     auto tableEntry = getTableCatalogEntry(tableID);
-    tableNameToIDMap.erase(tableEntry->getName());
-    tableNameToIDMap.emplace(newName, tableID);
+    if (tableEntry->getType() == CatalogEntryType::RDF_GRAPH_ENTRY) {
```
Comment (acquamarin) [inline src/catalog/catalog_content.cpp:193]: use switch?

**Review by acquamarin — APPROVED** [review]

File: src/storage/wal/wal.cpp (hunk 1 above)
Comment (ray6080) [reply src/storage/wal/wal.cpp:158]: I asserted before the check. But kept the code here for now under release mode.

File: src/include/binder/binder.h (hunk 2 above)
Comment (ray6080) [reply src/include/binder/binder.h:19]: ah yeah. I was thinking to remove the include of `catalog.h` (but didn't end up doing it). that's why added extra forward declarations here. removed `catalog.h` now.

File: src/include/catalog/catalog_content.h (hunk 4 above)
Comment (ray6080) [reply src/include/catalog/catalog_content.h:85]: they are indeed private. just want to separate them from other functions used in `Catalog`. the comment is confusing though. I removed it separate them with another `private` keyword.

File: src/catalog/catalog_content.cpp (hunk 5 above)
Comment (ray6080) [reply src/catalog/catalog_content.cpp:80]: good catch.

**Pull request merged**